### ISSUE TITLE
Unit stacking bug with Princess deployment

### ISSUE THREAD

**Issue by monbvol** [MegaMek/megamek#443]
[gamelog.zip](https://github.com/MegaMek/megamek/files/659992/gamelog.zip)
![stackingbug](https://cloud.githubusercontent.com/assets/17420975/21300222/f2a4d6f6-c556-11e6-9ab2-741e71d16efa.png)
[stackingbug.sav.gz](https://github.com/MegaMek/megamek/files/659993/stackingbug.sav.gz)

No error in log but included log anyway.

Quickdraw and Dervish got deployed to the same hex by the bot.  Lower left corner of the included screen shot.

No custom units in the match.

**Comment by pheonixstorm** [comment]
Is this still an issue?

**Comment by SuperStucco** [comment]
Yes, it is as of 49.12.  Princess is not checking for stacking limits with opposing units in overlapping deployment zones. This may be a double blind/sensor contact rules thing.  It doesn't violate stacking with it's own units.

**Comment by repligator** [comment]
As of 50.00, Princess _will not_ deliberately deploy her units on top of each other (beyond stacking limit). However, if there is an enemy unit she may deploy on top of them, which may cause one of her units to domino effect itself on top of one of her other units. In the attached save file, Princess has a force of 30 Wasps on a map with only 25 hexes. In both cases, she only deployed 25 of them.
[deploy_test.sav.gz](https://github.com/user-attachments/files/16256295/deploy_test.sav.gz)
![Screenshot_20240716_161451](https://github.com/user-attachments/assets/cce4ae72-604c-40ca-8335-aff24b0ad2be)
![Screenshot_20240716_162406](https://github.com/user-attachments/assets/3363a71d-700a-4f1f-b88d-31994164989c)

**Comment by Sleet01** [comment]
Sounds like we already have the correct handling, we just need to remove from the deployment candidates list any hexes that contain enemies that can't share occupation, in the same way that (presumably) we do for friendlies.

I will say, though, that this is such an edge case that I'm not sure it warrants spending more time on.

**Comment by repligator** [comment]
> I will say, though, that this is such an edge case that I'm not sure it warrants spending more time on.

I'm inclined to agree. It's only going to occur if:

1. Double Blind is on
2. The deployment zones overlap (and there is a setting to warn the player of this).
3. All the initiative rolls and settings line up absolutely perfectly.

**Comment by KyleM5** [comment]
This is still occuring. 
Version 0.50.02
System: Windows 10
Java Vendor: Oracle Corporation     Java Version: 17.0.12
[4.sav.gz](https://github.com/user-attachments/files/18386959/4.sav.gz)
If you load the file, the AI will occasionally stack some units, occasionally do it right, and occasionally not deploy all of their units.

**Comment by Scoppio** [comment]
@Sleet01 I am unable to reproduce the event after doing 8 retries, I believe you might have fixed it.

**Comment by Sleet01** [comment]
We haven't had any further reports, so I am going to close this.
